src/org/exist/util/Configuration.java

```diff
@@ -837,17 +837,33 @@ private void configureBackend( final Optional<Path> dbHome, Element con ) throws
             }
         }
 
-        String collectionCache = getConfigAttributeValue( con, CollectionCacheManager.CACHE_SIZE_ATTRIBUTE );
+        String collectionCache = getConfigAttributeValue(con, CollectionCacheManager.CACHE_SIZE_ATTRIBUTE);
+        if(collectionCache != null) {
+            collectionCache = collectionCache.toLowerCase();
 
-        if( collectionCache != null ) {
+            try {
+                final int collectionCacheBytes;
+                if(collectionCache.endsWith("k")) {
+                    collectionCacheBytes = 1024 * Integer.valueOf(collectionCache.substring(0, collectionCache.length() - 1));
+                } else if(collectionCache.endsWith("kb")) {
+                    collectionCacheBytes = 1024 * Integer.valueOf(collectionCache.substring(0, collectionCache.length() - 2));
+                } else if(collectionCache.endsWith("m")) {
+                    collectionCacheBytes = 1024 * 1024 * Integer.valueOf(collectionCache.substring(0, collectionCache.length() - 1));
+                } else if(collectionCache.endsWith("mb")) {
+                    collectionCacheBytes = 1024 * 1024 * Integer.valueOf(collectionCache.substring(0, collectionCache.length() - 2));
+                } else if(collectionCache.endsWith("g")) {
+                    collectionCacheBytes = 1024 * 1024 * 1024 * Integer.valueOf(collectionCache.substring(0, collectionCache.length() - 1));
+                } else if(collectionCache.endsWith("gb")) {
+                    collectionCacheBytes = 1024 * 1024 * 1024 * Integer.valueOf(collectionCache.substring(0, collectionCache.length() - 2));
+                } else {
+                    collectionCacheBytes = Integer.valueOf(collectionCache);
+                }
 
-            if( collectionCache.endsWith( "M" ) || collectionCache.endsWith( "m" ) ) {
-                collectionCache = collectionCache.substring( 0, collectionCache.length() - 1 );
-            }
+                config.put(CollectionCacheManager.PROPERTY_CACHE_SIZE_BYTES, collectionCacheBytes);
```

Comment:
@shabanovd I did what I thought was correct and best. So my mistake. Feel free to improve on my work